### PR TITLE
etcdserver: Error handling for invalid empty raft cluster 

### PR DESCRIPTION
**Pull request by sinsharat**
TODO implemented for GetClusterFromRemotePeers should not return nil error with an invalid empty cluster

### CODE REVIEW

**Comment by sinsharat** [comment]
@xiang90  please review.

File: etcdserver/cluster_util.go [hunk 1]
```diff
@@ -94,7 +94,16 @@ func getClusterFromRemotePeers(urls []string, timeout time.Duration, logerr bool
...
 			continue
 		}
-		return membership.NewClusterFromMembers("", id, membs), nil
+		// check the length of membership members
+		// if the membership members are present then prepare and return raft cluster
+		// if membership members are not present then the raft cluster formed will be
+		// an invalid empty cluster hence return an no raft cluster member error
+		if len(membs) > 0 {
+			return membership.NewClusterFromMembers("", id, membs), nil
+		} else {
```
Comment (xiang90) [inline etcdserver/cluster_util.go]: remove this else statement.

File: etcdserver/cluster_util.go [hunk 2]
```diff
@@ -94,7 +94,16 @@ func getClusterFromRemotePeers(urls []string, timeout time.Duration, logerr bool
...
+			return nil, fmt.Errorf("could not detect raft cluster member(s) from the urls.")
```
Comment (xiang90) [inline etcdserver/cluster_util.go]: failed to get raft cluster member(s) from the given urls.

**Comment by xiang90** [comment]
fix the nits and squash commits into one? Then LGTM. Thanks for the contribution!

File: etcdserver/cluster_util.go (hunk 1 above)
Comment (sinsharat) [inline etcdserver/cluster_util.go]: ok i will remove it.

File: etcdserver/cluster_util.go (hunk 2 above)
Comment (sinsharat) [inline etcdserver/cluster_util.go]: @xiang90  you asked to remove else statement. i am confused whether you want me to remove else or correct the error message as specified by you. Let me know and i will correct it and will re-submit after doing corrections.

Comment (xiang90) [inline etcdserver/cluster_util.go]:
```go
		if len(membs) > 0 {
			return membership.NewClusterFromMembers("", id, membs), nil
		}

		return nil, fmt.Errorf("failed to get raft cluster member(s) from the given urls.")
```

Comment (sinsharat) [inline etcdserver/cluster_util.go]: ok will update the error statement as per your suggestion. Thanks for providing your valuble feedback.

**Comment by sinsharat** [comment]
@xiang90  i have done the changes as per your comments. let me know if any other changes required.

**Comment by xiang90** [comment]
There are 9 commits inside this pr. We need to squash them down to one.

**Pull request closed without merging**